doc/west/repo-tool.rst

```diff
@@ -0,0 +1,265 @@
+
+The rest of this page introduces multi-repo concepts and gives an overview of
+the multi-repo commands, along with an example workflow.
+
+.. note::
+
+   The multi-repo commands are meant to augment Git in minor ways for
+   multi-repo work, not replace it. For tasks that aren't multi-repo-related,
+   use plain Git commands.
+
+   This page tries to make it transparent what the multi-repo commands do
+   behind the scenes.
+
+Manifests
+=========
+
+A *manifest* is a `YAML <http://yaml.org/>`_ file that gives the URL and
+revision for each repository in the multi-repo (each *project*), possibly along
+with other information. Manifests are normally stored in a separate Git
+repository.
+
+Running ``west init`` to initialize a West installation will clone the selected
+manifest repository. Running ``west init`` without specifying a manifest
+repository will use `a default manifest repository
+<https://github.com/zephyrproject-rtos/manifest>`_. A different manifest
+repository can be selected with ``west init --manifest-url URL``.
+
+The format of the West manifest is described by a `pykwalify
+<https://pypi.org/project/pykwalify/>`_ schema, found `here
+<https://github.com/zephyrproject-rtos/zephyr/blob/master/scripts/meta/west/commands/schema.yml>`_.
+
+The ``manifest-rev`` branch
+***************************
+
+Each project managed through the multi-repo commands has a ``manifest-rev``
+branch that points to the project's revision as specified in the manifest. The
+``manifest-rev`` branch is updated whenever project data is fetched (see the
+`command overview`_ below).
+
+All branches created through the multi-repo commands track the ``manifest-rev``
+branch. Several multi-repo commands also use ``manifest-rev`` as a reference
+for the upstream revision (as of the most recent fetch).
+
+.. note::
+
+   ``manifest-rev`` is a normal Git branch, and is only treated specially by
+   name. If you delete it or otherwise modify it, it will be
+   recreated/reset when upstream data is next fetched by ``west``, as if
+   through ``git reset --hard`` (though ``git update-ref`` is used internally).
+
+   Since ``manifest-rev`` represents the upstream revision as of the most
+   recent fetch, it is normally a bad idea to modify it.
+
+   ``manifest-rev`` was added to allow branches to track SHA revisions, and to
+   give a consistent reference for the upstream revision regardless of how the
+   manifest changes over time.
+
+Command overview
+================
+
+This section gives a quick overview of the multi-repo commands, split up by
+functionality. All commands loosely mimic the corresponding Git command, but in
+a multi-repo context (``west fetch`` fetches upstream data without updating
+local work branches, etc.)
+
+Passing no projects to commands that accept a list of projects usually means to
+run the command for all projects.
+
+.. note::
+
+   For the most up-to-date documentation, see the command help texts (e.g.
```

Comment:
We **could** pull the help content from the west python doc strings so it would automatically show up here... via http://www.sphinx-doc.org/en/master/usage/extensions/autodoc.html

Reply:
Agreed, let's have the simple overview here